src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -28,7 +33,6 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
   val anonymousTypeArguments = builder.anonymousTypeArguments
   val kdoc = builder.kdoc.build()
   val annotations = builder.annotations.toImmutableList()
-  val modifiers = builder.modifiers.toImmutableSet()
```

Comment:
We still need to expose this (as Set). Delegate?

Reply:
Actually we need a defensive copy of the kind or to make it immutable.

Reply:
It shouldn't be possible to modify the instance as all fields are internal, but I can look into improving encapsulation. `sealed` doesn't work with `data`, right?

Reply:
One builder creating two classes where the second adds a modifier will see that reflected on both.

Sealed works with data.